src/arena.c

```diff
@@ -1282,15 +1293,9 @@ arena_dalloc_promoted(tsdn_t *tsdn, void *ptr, tcache_t *tcache,
 		 * Currently, we only do redzoning for small sampled
 		 * allocations.
 		 */
-		assert(bumped_usize == SC_LARGE_MINCLASS);
 		safety_check_verify_redzone(ptr, usize, bumped_usize);
 	}
-	if (bumped_usize <= tcache_maxclass && tcache != NULL) {
-		tcache_dalloc_large(tsdn_tsd(tsdn), tcache, ptr,
-		    sz_size2index(bumped_usize), slow_path);
-	} else {
-		large_dalloc(tsdn, edata);
-	}
+	large_dalloc(tsdn, edata);
```

Comment:
I was running into issues when allowing the thread cache to keep these oddly sized edatas around, since they don't necessarily adhere to the invariants maintained elsewhere. Is it worth complicating the tcache logic to handle this, instead of just not caching `edatas` from sampled allocations?

Reply:
Probably fine to not cache these. Question, we are going through the large pipeline, because these edatas are sampled and not slabs, with the only difference from large is their size is not a large size class. Is that right? I just realized that some of my previous comments were probably wrong and missed this part.

Reply:
> We are going through the large pipeline, because these edatas are sampled and not slabs, with the only difference from large is their size is not a large size class. Is that right?

Yes, this was precisely my intention, so that we don't have to break the previous invariants around size-classes throughout the rest of the codebase, and instead isolate the special-casing to the small number of sites modified in this PR.